core/src/main/scala/cr/pulsar/Producer.scala

```diff
@@ -62,17 +64,15 @@ object Producer {
   /**
     * It creates a simple [[Producer]] with the supplied options.
     */
-  def withOptions[
-      F[_]: Concurrent: ContextShift: Parallel,
-      E: Inject[*, Array[Byte]]
-  ](
+  def withOptions[F[_]: Concurrent: ContextShift: Parallel, E](
       client: Pulsar.T,
       topic: Topic,
+      schema: Schema[E],
```

Comment:
Now `Schema` becomes a mandatory argument for this constructor. Is this intended? Shouldn't we actually make it part of `Options`?

Reply:
That was intentional, yes. Hmm no, either you use schemas or you don't, unless you want to duplicate code.

Reply:
Let's have `BYTES` schema as a default parameter then. 

```suggestion
      schema: Schema[E] = Schema.BYTES,
```

Reply:
I changed the design of this, I made it implicit. Still testing, will push changes as soon as I have it working